### DISCUSSION TITLE
Mayo::ui::WidgetMainControl not found

### DISCUSSION THREAD

**Discussion by MattiaMontanari** [fougue/mayo#302] [Q&A]
I'm trying to add a button in the widge_main_control.ui but I get an error, seems like some #include directives are not setup correctly. I place the button and once I right-click -> Go to slot.. I get the error shown in the bottom-right corner of the screenshot below.

<img width="1259" alt="image" src="https://github.com/user-attachments/assets/1d592941-e567-46f7-bd4f-88ab2e83284a">

I'm on MacOS and I just opened the CMakeFileLists from QtCreator

**Comment by MattiaMontanari** [comment]
> I'm trying to add a button in the widge_main_control.ui but I get an error, seems like some #include directives are not setup correctly. I place the button and once I right-click -> Go to slot.. I get the error shown in the bottom-right corner of the screenshot below.
> 
> <img alt="image" width="1259" src="https://private-user-images.githubusercontent.com/13785944/372363880-1d592941-e567-46f7-bd4f-88ab2e83284a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3Njg1ODEsIm5iZiI6MTcyNzc2ODI4MSwicGF0aCI6Ii8xMzc4NTk0NC8zNzIzNjM4ODAtMWQ1OTI5NDEtZTU2Ny00NmY3LWJkNGYtODhhYjJlODMyODRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDA3MzgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzMWFlOWE2NzcyNzVkNDYwODQ3NWMxMDUzZTVmZGYzMjc1MTdkN2ZhNjRjYmQ3NzFlYzZlOGZhYmJmYTFmMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._hlrZ5nWcrQETALoPJAN_fqEhBpvOnsO031XiSE65R4">
> I'm on MacOS and I just opened the CMakeFileLists from QtCreator

After some Googling it seem a setup issue. On [Windows this problem may not appear](Ui::MainWindow)

**Reply by HuguesDelorme** [reply]
I have tried on Windows within QtCreator and got not error
Maybe worth trying in standalone QtDesigner application if you get the same issue
